### PR TITLE
workflows: `issue_comment` triggers refactoring

### PR DESCRIPTION
**Pull request by nbusseneau**
Please review per commit :)

### CODE REVIEW

**Review by aanm — APPROVED** [review]

File: .github/workflows/conformance-aks-v1.10.yaml
Comment (twpayne) [inline .github/workflows/conformance-aks-v1.10.yaml]: Doing an exact match here will break if there's a newline after `/ci-aks-1.0`. Can we use a regexp match instead?

Comment (twpayne) [reply .github/workflows/conformance-aks-v1.10.yaml]: It seems that [there's no regexp matching available](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions) 😢

Comment (twpayne) [reply .github/workflows/conformance-aks-v1.10.yaml]: As `/test` is the problem, how about for `/test` only:

```
(startsWith(github.event.comment.body, '/test') && !startsWith(github.event.comment.body, '/test-'))
```

it's a hack, but is resistant to new lines and comments after the `/command`.

Comment (nbusseneau) [reply .github/workflows/conformance-aks-v1.10.yaml]: Yea, as said in the commit message my main goal was getting rid of the "double testing" issue, with this limitation as side-effect. Unfortunately GitHub Actions does not support filtering strings using regex (see [here](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions) for what's possible in native syntax), so even if we wanted to we couldn't do it from within the `if` context. Would have to kludge around with a shell step actively failing the job if not matching...

Comment (nbusseneau) [reply .github/workflows/conformance-aks-v1.10.yaml]: You posted so fast GH API did not have time to show me the messages, wow! :D
OK let's ship it.

Comment (nbusseneau) [reply .github/workflows/conformance-aks-v1.10.yaml]: PTAL.

**Comment by nbusseneau** [comment]
test-nbu

**Comment by nbusseneau** [comment]
test-nbu

**Comment by nbusseneau** [comment]
test-nbu

**Comment by nbusseneau** [comment]
test-nbu

**Comment by nbusseneau** [comment]
test-nbu

**Review by twpayne — APPROVED** [review]
LGTM

We also need to add docs, but that can be in a follow-up PR.

**Comment by nbusseneau** [comment]
> We also need to add docs, but that can be in a follow-up PR.

Too late, I was already doing it :D

File: .github/maintainers-little-helper.yaml
Comment (nbusseneau) [inline .github/maintainers-little-helper.yaml:53]: @aanm Will that work as intended?

**Review by christarazi — APPROVED** [review]

File: .github/maintainers-little-helper.yaml
Comment (aanm) [reply .github/maintainers-little-helper.yaml:53]: @nbusseneau yes but the right regex would be `(^(/test|test-me-please))` otherwise `/test-me-please` will also be considered.

Comment (nbusseneau) [reply .github/maintainers-little-helper.yaml:53]: Yes, I thought it was nice having it in during the transition phase. I also used that same regex allowing both on Jenkins.

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Review by twpayne — APPROVED** [review]

File: .github/workflows/conformance-aws-cni.yaml
Comment (pchaigno) [inline .github/workflows/conformance-aws-cni.yaml]: If I'm reading this correctly, it's a change in behavior. `ci-awscni` now won't trigger the test if only `test/` or `Documentation/` were modified. That was done on purpose, as I'm assuming someone writing `ci-awscni` explicitly know what they are doing.

That's useful for quick tests where you only want to send a nop change and trigger the tests.

File: .github/workflows/conformance-aks.yaml
Comment (pchaigno) [inline .github/workflows/conformance-aks.yaml]: Is this required for `test-me-please`? We don't have `test-me-please-1.10`, do we?

**Review by pchaigno — CHANGES_REQUESTED** [review]
One concern below, the other is a nit. Also, where was this tested?

File: .github/workflows/conformance-aks.yaml
Comment (nbusseneau) [reply .github/workflows/conformance-aks.yaml]: Not required, just added via grep replace. Decided to leave it because it has no side-effects and will be removed anyway as this is only for the transition period.

File: .github/workflows/conformance-aws-cni.yaml
Comment (nbusseneau) [reply .github/workflows/conformance-aws-cni.yaml]: Indeed, I didn't think this behaviour was useful so I erred on the side of simplifying the conditions for consistency. If we think it is, I can add it back in :)

Comment (pchaigno) [reply .github/workflows/conformance-aws-cni.yaml]: Unless it's adding too much complexity, yes please :pray:

Comment (nbusseneau) [reply .github/workflows/conformance-aws-cni.yaml]: PTAL.

Comment (pchaigno) [reply .github/workflows/conformance-aws-cni.yaml]: I'm not seeing any change :detective: Was it adding too much complexity?

Comment (nbusseneau) [reply .github/workflows/conformance-aws-cni.yaml]: Uuuh, is the GitHub cache playing tricks on you? 👀 Anyway, the changes are available here, from the force push link below: https://github.com/cilium/cilium/compare/f8baa22cbe9fb6a943f7bdfbf04dc31ae0ca349c..e44285b0bfc7c423921b07460e0a58461fecd7ec

Comment (pchaigno) [reply .github/workflows/conformance-aws-cni.yaml]: Uh, why are they in the documentation commit? Might be worth updating the commit title at least.

**Review by pchaigno — APPROVED** [review]
@nbusseneau Did you validate this with a test commit already?

File: .github/workflows/conformance-aws-cni.yaml
Comment (nbusseneau) [reply .github/workflows/conformance-aws-cni.yaml]: Oh are they? I mixed up the fixup then.

Comment (nbusseneau) [reply .github/workflows/conformance-aws-cni.yaml]: I have reworked the commits so that the changes are in the proper place.

**Comment by nbusseneau** [comment]
> @nbusseneau Did you validate this with a test commit already?

I looked at the PR history and was like "hmmm, no I did not, why didn't I?" 🤔 So I added a `DO NOT MERGE` commit and then remembered why I did not: this PR is from my fork, and cannot trigger test runs.

I am opening a separate PR for testing.

**Comment by nbusseneau** [comment]
Anybody knows why https://github.com/cilium/cilium/actions/runs/1270095584 is failing? I suppose it is due to this:
```
{[3/5] update MLH config with new `/test` trigger}
  Running on 417c7794d8797687c82bfe31abbe3a1cba8f8258
  Warning: WARNING:COMMIT_MESSAGE: Missing commit description - Add an appropriate one
  
  
  NOTE: For some of the reported defects, checkpatch may be able to
        mechanically convert to the typical style using --fix or --fix-inplace.
  
  "[PATCH] update MLH config with new `/test` trigger" has style problems, please review.
  
  NOTE: Ignored message types: C99_COMMENT_TOLERANCE COMMIT_LOG_LONG_LINE COMPLEX_MACRO CONSTANT_CONVERSION CONST_STRUCT EMAIL_SUBJECT FILE_PATH_CHANGES FROM_SIGN_OFF_MISMATCH GIT_COMMIT_ID JIFFIES_COMPARISON LEADING_SPACE LONG_LINE_COMMENT MACRO_WITH_FLOW_CONTROL MULTISTATEMENT_MACRO_USE_DO_WHILE NOT_UNIFIED_DIFF PRINTK_WITHOUT_KERN_LEVEL TRAILING_SEMICOLON TRAILING_STATEMENTS VOLATILE
  
  NOTE: If any of the errors are false positives, please report
        them to the maintainer, see CHECKPATCH in MAINTAINERS.
```
Do we not allow empty commit descriptions anymore?

**Comment by nbusseneau** [comment]
All tests in #17465 successfully triggered, however please note that the PR tests the `pull_request` path and not the `issue_comment` triggers themselves, so this test only guarantees that the PR changes are syntactically correct and won't break workflows after merge.

**Comment by pchaigno** [comment]
> Do we not allow empty commit descriptions anymore?

cc @qmonnet

**Comment by pchaigno** [comment]
Reviews are in and required tests are passing. Marking ready to merge.

**Comment by qmonnet** [comment]
> Do we not allow empty commit descriptions anymore?

No.
There was a Slack discussion about that some time ago, and I made checkpatch complain on empty commit logs as a result. Only my checkpatch change introduced a bug and checkpatch would not make the GitHub action fail when detecting errors :disappointed:. Fixed now, so normal checkpatch activity should resume, with complaints on empty commit logs in addition to the previous reports.

**Comment by pchaigno** [comment]
> There was a Slack discussion about that some time ago, and I made checkpatch complain on empty commit logs as a result.

IIRC, the conclusion of that discussion was that empty commit are sometimes useful, as illustrated here. I think we should revert.

**Pull request merged**
